File: src/test/java/org/stt/config/PasswordSettingTest.java
package org.stt.config;

import org.junit.Test;

import java.io.UnsupportedEncodingException;

import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.CoreMatchers.not;
import static org.junit.Assert.assertThat;

public class PasswordSettingTest {
    @Test
    public void shouldEncryptAndDecryptPassword() throws UnsupportedEncodingException {
        // GIVEN
        byte[] originalPassword = "Hello World".getBytes("UTF-8");

        // WHEN
        PasswordSetting passwordSetting = PasswordSetting.fromPassword(originalPassword);

        // THEN
        assertThat(passwordSetting.getPassword(), is(originalPassword));
    }

    @Test
    public void shouldNotStorePlainPassword() throws UnsupportedEncodingException {
        // GIVEN
        byte[] originalPassword = "Hello World".getBytes("UTF-8");

        // WHEN
        PasswordSetting passwordSetting = PasswordSetting.fromPassword(originalPassword);

        // THEN
        assertThat(passwordSetting.encodedPassword, is(not(originalPassword)));
    }

}

File: src/test/kotlin/org/stt/config/PasswordSettingTest.kt
package org.stt.config

import org.assertj.core.api.Assertions.assertThat
import org.junit.Test
import java.nio.charset.StandardCharsets

class PasswordSettingTest {
    @Test
    fun shouldEncryptAndDecryptPassword() {
        // GIVEN
        val originalPassword = "Hello World".toByteArray(StandardCharsets.UTF_8)

        // WHEN
        val passwordSetting = PasswordSetting.fromPassword(originalPassword)

        // THEN
        assertThat(passwordSetting.password).isEqualTo(originalPassword)
    }

    @Test
    fun shouldNotStorePlainPassword() {
        // GIVEN
        val originalPassword = "Hello World".toByteArray(StandardCharsets.UTF_8)

        // WHEN
        val passwordSetting = PasswordSetting.fromPassword(originalPassword)

        // THEN
        assertThat(passwordSetting.encodedPassword).isNotEqualTo(originalPassword)
    }

}